File: app/src/main/java/com/musicplayer/OpenMusic/data/base/SongDao.java
package com.musicplayer.OpenMusic.data.base;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.musicplayer.OpenMusic.data.Song;

import java.util.List;

@Dao
public interface SongDao {
    @Insert
    void insert(Song song);

    @Insert
    void insertAll(List<Song> song);

    @Query("SELECT * FROM Song ORDER BY song_path")
    List<Song> getAll();

    @Query("DELETE FROM Song")
    void clearAll();

    @Delete
    void delete(Song song);

    @Query("UPDATE Song SET album_id=:albumId WHERE song_id=:songID")
    void setAlbum(String songID, String albumId);
}

File: app/src/main/java/com/musicplayer/OpenMusic/data/base/SongDao.kt
package com.musicplayer.OpenMusic.data.base

import androidx.room.Dao
import com.musicplayer.OpenMusic.data.Song
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface SongDao {
    @Insert
    fun insert(song: Song)

    @Insert
    fun insertAll(song: ArrayList<Song>)

    @get:Query("SELECT * FROM Song ORDER BY song_path")
    val all: ArrayList<Song>

    @Query("DELETE FROM Song")
    fun clearAll()

    @Delete
    fun delete(song: Song)

    @Query("UPDATE Song SET album_id=:albumId WHERE song_id=:songID")
    fun setAlbum(songID: String, albumId: String)
}